### PR TITLE
Scan JS Package: Remove Build

### PR DESCRIPTION
**Pull request by nateweller**
## Proposed changes:

* Removes references to a "build" directory from the `package.json` file.
* Removes use of `jetpack:src` from `exports`, since `default` is now the same `./src/index.ts` value.
* Update `composer.json`.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

p1732032090059999-slack-CDLH4C1UZ

peb6dq-36i-p2

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

* Ensure all builds are passing.
* Run `cd projects/js-packages/storybook && npm run storybook:dev`
* Open `http://localhost:50240/?path=/docs/js-packages-components-threat-fixer-button--docs`
* Validate that the component renders (uses methods from the Scan package).

Screenshot 2024-11-19 at 1 53 42 PM

### CODE REVIEW

File: projects/js-packages/scan/composer.json [hunk 1]
```diff
@@ -18,16 +18,12 @@
 			"pnpm run build"
 		],
 		"build-production": [
-			"NODE_ENV=production BABEL_ENV=production pnpm run build"
-		],
-		"watch": [
-			"Composer\\Config::disableProcessTimeout",
-			"pnpm run watch"
+			"pnpm run build"
```
Comment (anomiex) [inline projects/js-packages/scan/composer.json]: I'd probably keep this as it was, even though the flags shouldn't make a difference anymore.
```suggestion
			"NODE_ENV=production BABEL_ENV=production pnpm run build"
```

**Review by anomiex — COMMENTED** [review]
🤷 This improves on the current situation. OTOH, it makes the package less usable for people outside the monorepo (cf. https://github.com/Automattic/jetpack/issues/39532). I don't know whether that's a goal for this package.

**Comment by nateweller** [comment]
> I don't know whether that's a goal for this package.

@anomiex Right now, the Scan/Security team's intention is primarily to provide a means of sharing code across the main Jetpack plugin and Jetpack Protect. Although if the package was more easily consumable outside of the monorepo, we could use it in Calypso or other places in the future, which sounds nice.

I suppose the other direction would be to build the build, and distribute the JS. I'm not sure exactly what that would mean in terms of the resulting bundling/affects on deps and peer deps, if anything. I can explore that more and draft an alternate PR testing out that approach. :+1:

**Comment by anomiex** [comment]
> I suppose the other direction would be to build the build, and distribute the JS. I'm not sure exactly what that would mean in terms of the resulting bundling/affects on deps and peer deps, if anything.

Within the monorepo it shouldn't change anything, since stuff within the monorepo should continue to use the "jetpack:src" entries.

Externally it'd bring back a `build/` dir similar to [what was there](https://github.com/Automattic/jetpack-scan/tree/50810994a6482e5923d9f39888d6551f1ea9bbf2/build) before #39754. The build setup there with `tsc` didn't actually bundle anything, it just turns the TypeScript code into `.js` + `.d.ts` that's easier for external consumers to deal with.

I don't see any relevant dependency changes in/since #39754, so that should be fine. But you'll have to resolve whatever errors led to 377e8739b2ceb1ac53f2fee2403c1b991547298f in a different manner. If the problem was stuff like
```
[plugins/jetpack]     Cannot find module './types/index.js' from '../../js-packages/scan/src/index.ts'
```
from jest tests then this might help:
<details><summary>patch</summary>

```diff
diff --git a/tools/js-tools/package.json b/tools/js-tools/package.json
index 784f44cf8d..43c9ca9d81 100644
--- a/tools/js-tools/package.json
+++ b/tools/js-tools/package.json
@@ -56,6 +56,7 @@
                "sort-package-json": "1.50.0",
                "svelte": "4.2.19",
                "svelte-eslint-parser": "0.39.2",
+               "ts-jest-resolver": "2.0.1",
                "typescript": "5.0.4",
                "yaml": "2.2.2"
        }
diff --git a/tools/js-tools/jest/jest-resolver.js b/tools/js-tools/jest/jest-resolver.js
index 597c41156b..03353b69e7 100644
--- a/tools/js-tools/jest/jest-resolver.js
+++ b/tools/js-tools/jest/jest-resolver.js
@@ -1,3 +1,5 @@
+const tsJestResolver = require( 'ts-jest-resolver' );
+
 // Some packages assume that a "browser" environment is esm or otherwise break in node.
 // List them here and the resolver will adjust the conditions to resolve them as "node" instead.
 // cf. https://github.com/microsoft/accessibility-insights-web/pull/5421#issuecomment-1109168149
@@ -20,7 +22,7 @@ module.exports = ( path, options ) => {
                conditions.add( 'node' );
        }
 
-       return options.defaultResolver( path, {
+       return tsJestResolver( path, {
                ...options,
                basedir,
                conditions,
```

</details>

**Comment by nateweller** [comment]
Closing this in favor of https://github.com/Automattic/jetpack/pull/40299

**Pull request closed without merging**